src/julia/tests/test_tools.py

```diff
@@ -0,0 +1,79 @@
+import pytest
+import platform
+import sysconfig
+import os
+import glob
+
+from julia.tools import julia_py_executable
+
+
+def julia_py_with_command_extension():
+    extension = ".cmd" if platform.system() == "Windows" else ""
+    return "julia-py" + extension
+
+
+def glob_mock(path):
+    # we're only handling the case when the glob is ".../julia-py*" or just "julia-py*"
+    # if dirname(path) is empty then return empty list - this is indicator that we shouldn't "find" any files matching the glob
+    return [] if os.path.dirname(path) == "" else [os.path.join(os.path.dirname(path), julia_py_with_command_extension())]
+
+
+def test_find_julia_py_executable_by_scheme(monkeypatch):
+    # could extend this to test different kinds of scheme
+    # right now we just fake the "posix_user" scheme and don't use the rest of get_path_mock_scheme()
```

Comment:
Let's not add code that we don't use. I think you can just patch `get_scheme_names` and make sure that it always returns `"posix_user"` and then `assert` it in `get_path_mock_scheme`.

Reply:
Great, sure. I wanted to illustrate what's available in case needed. I've cut it down.